optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/rotation/RotationPage.java

```diff
@@ -84,46 +86,66 @@ public void onTenantChanged(final @Observes TenantStore.TenantChange tenant) {
-                    .collect(groupingBy(Shift::getSpot));
+                                                                                     .collect(groupingBy(Shift::getSpot));
 
             viewport = rotationViewportFactory.getViewport(shiftsBySpot);
             viewportView.setViewport(viewport);
             return resolve();
         });
     }
 
-    private Promise<ShiftTemplate> fetchShiftTemplate() {
+    private Promise<Collection<ShiftTemplate>> fetchShiftTemplate() {
         return new Promise<>((resolve, reject) -> {
             ShiftRestServiceBuilder.getTemplate(tenantStore.getCurrentTenantId(), onSuccess(resolve::onInvoke));
         });
     }
 
-    private List<Shift> buildShiftList(final ShiftTemplate shiftTemplate) {
+    private List<Shift> buildShiftList(final Collection<ShiftTemplate> shiftTemplate) {
         final AtomicLong id = new AtomicLong(0L);
-        return shiftTemplate.getShiftList().stream()
-                .flatMap(shiftInfo -> shiftInfo.getSpotList().stream().map(spot -> newShift(id.getAndIncrement(), shiftInfo, spot)))
-                .collect(toList());
+        return shiftTemplate.stream()
+                            .map(shiftInfo -> newShift(id.getAndIncrement(), shiftInfo)).collect(toList());
     }
 
     private Shift newShift(final Long id,
-                           final ShiftInfo shift,
-                           final Spot spot) {
-
-        final TimeSlot newTimeSlot = new TimeSlot(
-                tenantStore.getCurrentTenantId(),
-                shift.getStartTime(),
-                shift.getEndTime());
-
+                           final ShiftTemplate shift) {
         final Shift newShift = new Shift(
-                tenantStore.getCurrentTenantId(),
-                spot,
-                newTimeSlot);
+                                         tenantStore.getCurrentTenantId(),
+                                         shift.getSpot(),
+                                         getStartDateTime(shift),
+                                         getEndDateTime(shift));
 
         newShift.setId(id);
-
         return newShift;
     }
 
+    public static LocalDate getBaseDate() {
+        return LocalDate.of(0, 1, 1);
+    }
+
+    public static OffsetDateTime getBaseDateTime() {
+        return OffsetDateTime.of(getBaseDate().atTime(LocalTime.MIDNIGHT), ZoneOffset.UTC);
+    }
+
+    private OffsetDateTime getStartDateTime(ShiftTemplate shift) {
+        return OffsetDateTime.of(getBaseDate()
+                                              .plusDays(shift.getOffsetStartDay())
+                                              .atTime(shift.getStartTime()), ZoneOffset.UTC);
+    }
+
+    private OffsetDateTime getEndDateTime(ShiftTemplate shift) {
+        return OffsetDateTime.of(getBaseDate()
+                                              .plusDays(shift.getOffsetEndDay())
+                                              .atTime(shift.getEndTime()), ZoneOffset.UTC);
```

Comment:
Formatting issues

Reply:
I am using KIE Formatter in cleanup, formatter and import order. It has an annoying feature where its forces method calls/the like to be aligned.

Reply:
Not sure what's happening. The IntelliJ formatter does not do that and they should work seamlessly together. Could you please double check your configurations? If it's really a formatter issue, would you mind updating it and sending a PR to `droolsjbpm-build-boostrap` repo? I think it's very important not to have this kind of noise on the PRs.